logicalplan/plan.go

```diff
@@ -247,7 +247,13 @@ func replacePrometheusNodes(plan parser.Expr) parser.Expr {
 	case *parser.StepInvariantExpr:
 		return &StepInvariantExpr{Expr: replacePrometheusNodes(t.Expr)}
 	case *parser.MatrixSelector:
-		return &MatrixSelector{VectorSelector: replacePrometheusNodes(t.VectorSelector), Range: t.Range, OriginalString: t.String()}
+		return &MatrixSelector{
+			VectorSelector: &VectorSelector{
+				VectorSelector: t.VectorSelector.(*parser.VectorSelector),
```

Comment:
did recursion break?

Reply:
I made `VectorSelector` a concrete type in MarixSelector so we would avoid casts in the rest of the codebase.

Reply:
Ah! Yeah makes sense!